plugins/de.cau.cs.kieler.klighd.lsp/src/de/cau/cs/kieler/klighd/lsp/interactive/layered/LayeredConstraintReevaluation.xtend

```diff
@@ -0,0 +1,316 @@
+
+        }
+        // Reevaluate the position constraints in the source and target layer accordingly
+        // Also examine the position constraint of the target node
+        if (posCons > 0 && posCons >= posIndexOfShifted) {
+            changedNodes.add(new ConstraintProperty(targetNode, LayeredOptions.CROSSING_MINIMIZATION_POSITION_CHOICE_CONSTRAINT, posCons - 1))
+        }
+
+    }
+
+    /**
+     * Offsets all nodes in a layer by {@code offset} that own a positional constraint that is greater or equal 
+     * than {@code startPos} and smaller than {@code endPos}.
+     * 
+     * @param layer The layer to examine as a list including its nodes ordered by position.
+     * @param The offset that should be applied on the position constraints
+     * 
+     */
+    def private offsetPosConstraintsOfLayerFromTo(List<KNode> layer, int offset, int startPos, int endPos,
+        KNode target) {
+
+        if (!layer.empty) {
+            for (var i = startPos; i < endPos + 1; i++) {
+                val node = layer.get(i)
+                val posChoiceCons = ConstraintsUtils.getPosConstraint(node)
+
+                if (node != target && posChoiceCons !== -1) {
+                    changedNodes.add(new ConstraintProperty(node, LayeredOptions.CROSSING_MINIMIZATION_POSITION_CHOICE_CONSTRAINT, posChoiceCons + offset))
+                }
+            }
+        }
+    }
+
+    /**
+     * Offsets all nodes in a layer by {@code offset} that own a positional constraint that is greater or equal 
+     * than {@code startPos}.
+     * 
+     * @param layer The layer to examine as a list including its nodes ordered by position.
+     * @param The offset that should be applied on the position constraints
+     * 
+     */
+    def private offsetPosConstraintsOfLayerFrom(List<KNode> layer, int offset, int startPos, KNode target) {
+        offsetPosConstraintsOfLayerFromTo(layer, offset, startPos, layer.length - 1, target)
+    }
+
+    /**
+     * Checks the constraints that are currently present in the model.
+     * This is necessary since the user is able to add an arbitrary number of constraints to the source code. 
+     * On this way they could define constraints that are obscure or harm the processing of the constraints.
+     * This should be caught before the layout is done. Constraints that are invalid are deactivated.
+     */
+    def static ckeckModelConstraints(List<KNode> nodes) {
+    }
+
+    /**
+     * Eliminates equal position constraints in the same layer since their semantics are unambitious.
+     * Parts of the reevaluation for adding a node are reused for this.
+     * This should be deactivated in default mode.
+     * @param nodes All nodes to examine. Requirement: The nodes must not have constraints that cause flat edges.
+     */
+    def static eliminateEqualPositionsInSameLayer(List<KNode> nodes) {
+
+        // TODO - still doesn't work completely
```

Comment:
TODO.

I am marking all these TODOs so they get looked at again, not all need to be fixed, but if they are important maybe they should or at least a ticket for them should be added.

Reply:
The method is unused. I suggest to remove it.